scylla/src/transport/iterator.rs

```diff
@@ -849,8 +851,8 @@ where
             let result = (self.fetcher)(paging_state).await?;
             let response = result.into_non_error_query_response()?;
             match response.response {
-                NonErrorResponse::Result(result::Result::Rows(mut rows)) => {
-                    paging_state = rows.metadata.paging_state.take();
+                NonErrorResponse::Result(result::Result::Rows(rows)) => {
+                    paging_state = rows.metadata().paging_state.clone();
                     let (proof, send_result) = self
```

Comment:
Because `metadata()` returns immutable reference. This is how @piodul designed it. I'm reluctant to do in-place modifications on metadata; OTOH, cloning page state does not involve allocations.

Reply:
Keep in mind that once paging state is decoupled from `ResultMetadata`, this is no longer going to be needed.